File: tests/fixtures/generic/clean/taint_xpathi_clean.js
// @audit-fixture
// @rule: taint_xpathi
// @category: security
// @expected: clean
const xpath = require('xpath');
const express = require('express');
const app = express();
app.get('/users', (req, res) => {
    const select = xpath.useNamespaces({});
    const nodes = select("//user[@name=$name]", doc, null, { name: req.query.name });
    res.json(nodes);
});

File: tests/fixtures/generic/clean/taint_xpathi_clean.cs
// @audit-fixture
// @rule: taint_xpathi
// @category: security
// @expected: clean
using System.Xml;
using System.Text.RegularExpressions;
public class XmlController : Controller {
    public IActionResult Search() {
        var name = Regex.Replace(Request.Query["name"].ToString(), @"[^\w]", "");
        var nodes = doc.SelectNodes("//user[@name='" + name + "']");
        return Ok(nodes?.Count);
    }
}
